presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSource.java

```diff
@@ -83,6 +83,12 @@ public OrcBatchPageSource(
         this(recordReader, orcDataSource, columns, typeManager, systemMemoryContext, stats, runtimeStats, nCopies(columns.size(), false));
     }
 
+    /**
+     * @param columns an ordered list of the fields to read
+     * @param isRowPositionList list of indices of columns. If true, then the column with the matching
```

Comment:
> list of indices of columns

This is a list of booleans, not indices, right?

This design is indeed strange.

Reply:
Can there be > 1 'row number' column? What does that mean?